tests/functional/extension_acp_test.php

```diff
@@ -244,4 +244,27 @@ public function test_actions()
 		$this->install_ext('vendor/moo');
 		$this->uninstall_ext('vendor/moo');
 	}
+
+	public function test_extensions_catalog()
+	{
+		// Access extensions catalog main page
+		$crawler = self::request('GET', 'adm/index.php?i=acp_extensions&mode=catalog&sid=' . $this->sid);
+		$this->assertContainsLang('ACP_EXTENSIONS_CATALOG', $this->get_content());
+
+		$this->assertContainsLang('BROWSE_EXTENSIONS_DATABASE', $crawler->filter('fieldset[class="quick quick-left"] > span > a')->eq(0)->text());
+		$this->assertContainsLang('SETTINGS', $crawler->filter('fieldset[class="quick quick-left"] > span > a')->eq(1)->text());
+
+		$form = $crawler->selectButton('Submit')->form();
+		$form['minimum_stability']->select('dev');
+		$crawler = self::submit($form);
+		$this->assertContainsLang('CONFIG_UPDATED', $crawler->filter('div[class="successbox"] > p')->text());
+
+		// Revisit extensions catalog main page after configuration change
+		$crawler = self::request('GET', 'adm/index.php?i=acp_extensions&mode=catalog&sid=' . $this->sid);
+		$this->assertContainsLang('ACP_EXTENSIONS_CATALOG', $this->get_content());
+
+		// Ensure catalog has 20 records (by default) in extensions list
+		$this->assertNotEmpty($crawler->filter('tbody > tr > td > strong')->eq(0)->text());
+		$this->assertNotEmpty($crawler->filter('tbody > tr > td > strong')->eq(19)->text());
```

Comment:
Why 20?

Reply:
Don't know, that's by design https://github.com/phpbb/phpbb/blob/c9484b7a1b4380d158edd5496ac583000e78a8f4/phpBB/includes/acp/acp_extensions.php#L696

Reply:
Wow, should'nt that be a configuration?

Reply:
Probably, but that's subject to another ticket/PR :)
Extension catalog requires a lot of such a work anyway, including catalog search etc. "must have" features.

Reply:
Indeed, just I got surprised of that sliced limit though.